system-int-test/test/cmr/system_int_test/search/autocomplete/suggestion_reindex_test.clj

```diff
@@ -90,29 +88,46 @@
                             (fu/processing-level-id "PL1")
                             {:DataCenters [(data-umm-spec/data-center {:Roles ["ARCHIVER"] :ShortName "DOI/USGS/CMG/WHSC"})]
                              :ScienceKeywords [(:ScienceKeywords (fu/science-keywords sk1 sk2))]})
+
         coll3 (d/ingest-concept-with-metadata-file "CMR-6287/C1000000029-EDF_OPS.xml"
-                                                         {:provider-id "PROV1"
-                                                          :concept-type :collection
-                                                          :format-key :echo10})
+                                                   {:provider-id "PROV1"
+                                                    :concept-type :collection
+                                                    :format-key :echo10})
+        _ (index/wait-until-indexed)
+        _ (index/reindex-suggestions)
         _ (index/wait-until-indexed)]
+    (f)))
+
+(use-fixtures :each (join-fixtures
+                     [(ingest/reset-fixture {"provguid1" "PROV1"})
+                      hu/grant-all-humanizers-fixture
+                      hu/save-sample-humanizers-fixture
+                      autocomplete-reindex-fixture]))
+
+(deftest reindex-suggestions-test
+  (testing "Ensure that response is in proper format and results are correct"
+    (compare-autocomplete-results
+     (get-in (search/get-autocomplete-json "q=level2") [:feed :entry])
+     [{:score 0.25604635 :type "organization" :value "Langley DAAC User Services" :fields "Langley DAAC User Services"}
+      {:score 0.041718055 :type "instrument" :value "lVIs" :fields "lVIs"}]))
+
+  (testing "Ensure science keywords are being indexed properly"
+    (are3
+     [query expected]
+     (let [actual (get-in (search/get-autocomplete-json query) [:feed :entry])]
+       (compare-autocomplete-results expected actual))          
+
+     "shorter match"
+     "q=solar"
+     [{:score 3.4995544, :type "science_keywords", :value "Solar Irradiance", :fields "Sun-Earth Interactions:Solar Activity:Solar Irradiance"}
+      {:score 1.9641972, :type "science_keywords", :value "Solar Irradiance", :fields "Atmosphere:Atmospheric Radiation:Solar Irradiance"}
+      {:score 0.20964509, :type "organization", :value "ACRIM SCF", :fields "ACRIM SCF"}
+      {:score 0.16771607, :type "organization", :value "Langley DAAC User Services", :fields "Langley DAAC User Services"}]
+
+     "more complete match"
+     "q=solar irradiation"
+     [{:score 5.0869484, :type "science_keywords", :value "Solar Irradiance", :fields "Sun-Earth Interactions:Solar Activity:Solar Irradiance"}
```

Comment:
I know this isn't part of this PR but can you explain why fields is constructed like it is? It looks like a colon separated path with `value` as the tail but why is this needed / why not a vector or a nested map?

Reply:
@CayvonH this was to provide EDSC values in a format they could use a single function to parse instead of checking for hierarchical vs flat suggestions. There is still some discussion over science keywords and the "fields" field